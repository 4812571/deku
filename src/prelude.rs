/*! Crate prelude

[What is a prelude?](https://doc.rust-lang.org/std/prelude/)
*/
pub use crate::{
    error::DekuError, DekuContainerRead, DekuContainerWrite, DekuRead, DekuUpdate, DekuWrite,
};
pub use bitvec::{
    order::BitOrder, order::Lsb0, order::Msb0, slice::BitSlice, vec::BitVec, view::BitView,
};
