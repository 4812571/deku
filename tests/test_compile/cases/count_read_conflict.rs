use deku::prelude::*;

#[derive(DekuRead)]
struct Test1 {
    #[deku(bits_read = "1", bytes_read = "2")]
    a: Vec<u8>,
}

#[derive(DekuRead)]
struct Test2 {
    #[deku(count = "1", bits_read = "1", bytes_read = "2")]
    a: Vec<u8>,
}

#[derive(DekuRead)]
struct Test3 {
    #[deku(count = "1", bits_read = "1")]
    a: Vec<u8>,
}

#[derive(DekuRead)]
struct Test4 {
    #[deku(count = "1", bytes_read = "2")]
    a: Vec<u8>,
}


fn main() {}
