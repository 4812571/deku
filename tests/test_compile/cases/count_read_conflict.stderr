error: conflicting: both `bits_read` and `bytes_read` specified on field
 --> $DIR/count_read_conflict.rs:5:24
  |
5 |     #[deku(bits_read = "1", bytes_read = "2")]
  |                        ^^^

error: conflicting: both `bits_read` and `bytes_read` specified on field
  --> $DIR/count_read_conflict.rs:11:37
   |
11 |     #[deku(count = "1", bits_read = "1", bytes_read = "2")]
   |                                     ^^^

error: conflicting: both `count` and `bits_read` specified on field
  --> $DIR/count_read_conflict.rs:17:20
   |
17 |     #[deku(count = "1", bits_read = "1")]
   |                    ^^^

error: conflicting: both `count` and `bytes_read` specified on field
  --> $DIR/count_read_conflict.rs:23:20
   |
23 |     #[deku(count = "1", bytes_read = "2")]
   |                    ^^^
